js/reference.js

```diff
@@ -253,8 +256,18 @@ var ERMrest = (function(module) {
              * @type {ERMrest.Reference}
              */
             get record() {
-                // TODO: remember these are copies of this reference.
-                return undefined;
+                var source = this._reference;
+                var newRef = _referenceCopy(source);
+                var columnOrders = source._table.columns._contextualize('record').all();
+
+                newRef._columns = [];
+                for (var i = 0; i < columnOrders.length; i++) {
+                    var column = columnOrders[i];
+                    if (source._columns.indexOf(column)) {
```

Comment:
This will not consider the column which is at index **0** in the `source._columns` array. Instead this statement should check for non **-1** index instead of positive index.

```javascript
if (source._columns.indexOf(column) != -1) {
```